### PR TITLE
Add Lattice tutorial

### PR DESCRIPTION
**Pull request by k-tamuraphys**
### Summary
Add a tutorial for Lattice module.
This demonstrates how to use the Lattice module in #365.

### CODE REVIEW

File: docs/tutorials/10_Lattice.ipynb
Comment (woodsp-ibm) [inline docs/tutorials/10_Lattice.ipynb:7]: Is there some more descriptive title (few more words rather than just one) that can improve on this. At a minimum I think it should be 'Lattices' plural with a s. Lattice singular does not feel right.

Now I see this tutorial shows to the point of creating second q ops. I will make a separate comment based on this and looking at the code, but I think we should show its actual usage with an algorithm to compute something so people have an example of how it would be used to do that.

**Comment by woodsp-ibm** [comment]
In regards of Lattice & Models being under 'problems'. I see the Model there can generate a second_q_ops but it does not extend BaseProblem, that defines this, whereby it could be given as  problem instance to say a ground state solver. Now BaseProblem is most likely not something that is applicable here since that seems to bring in drivers hence maybe some refactoring around that could be done. Maybe having this in Problem form and passable to the algorithms is something for later.  This is the additional comment I said I would make in seeing that the Lattice tutorial stops at the point of generating a second q ops and does not use it with any quantum algorithm to show some result.

**Comment by mrossinek** [comment]
Are there major differences between this tutorial and the one originally included in PR #365?

If not, then this is probably good to go once we merge that PR. I will review this after we have merged the above which will reduce the cluttered `Files` tab of this PR as it stands currently :+1:

**Comment by k-tamuraphys** [comment]
> Are there major differences between this tutorial and the one originally included in PR #365?
> 
> If not, then this is probably good to go once we merge that PR. I will review this after we have merged the above which will reduce the cluttered `Files` tab of this PR as it stands currently 👍

No, this is basically the same as what was previously included in PR #365., except for a few minor modifications.

**Comment by woodsp-ibm** [comment]
So once #365 is merged is the intent to remove all other files other than the tutorial. I guess there have been changes to these in #365 such that the copies here may now be different, such as renaming lattice folder to lattices etc.

**Comment by ikkoham** [comment]
Please merge this PR after #365.

**Review by mrossinek — APPROVED** [review]

**Comment by ikkoham** [comment]
Thank you. My Jupyter's black setting was set to 80 lines :cry:.

File: docs/tutorials/10_Lattice_Framework.ipynb
Comment (pbark) [inline docs/tutorials/10_Lattice_Framework.ipynb:15]:
```suggestion
    "In quantum physics (including condensed matter physics and high energy physics) we often study models on lattices. \n",
```

File: docs/tutorials/10_Lattice_Framework.ipynb
Comment (pbark) [inline docs/tutorials/10_Lattice_Framework.ipynb:754]:
```suggestion
    "We can generate the corresponding Hamiltonian of a given lattice using `FermiHubbardModel` class.\n",
```

File: docs/tutorials/10_Lattice_Framework.ipynb
Comment (pbark) [inline docs/tutorials/10_Lattice_Framework.ipynb:755]:
```suggestion
    "Here, we construct the Hamiltonian with uniform hopping and interaction parameters on a two-dimensional lattice."
```

File: docs/tutorials/10_Lattice_Framework.ipynb
Comment (pbark) [inline docs/tutorials/10_Lattice_Framework.ipynb:829]:
```suggestion
    "In this case, the weights of the lattice are regarded as the hopping matrix. After generating the Hamiltonian (`second_q_ops`) we can use a qubit converter to generate the qubit operators and/or use any of the available algorithms to solver the corresponding lattice problem."
```

**Review by pbark — CHANGES_REQUESTED** [review]

**Comment by pbark** [comment]
@ikkoham @k-tamuraphys it LGTM. I did some minor stylistic changes in the text around the tutorial. If you agree with these we can merge it.

**Comment by ikkoham** [comment]
Updated

**Review by pbark — APPROVED** [review]

**Review by mrossinek — APPROVED** [review]

**Pull request merged**
